src/cpu/aarch64/acl_gemm_convolution_utils.cpp

```diff
@@ -162,42 +156,106 @@ status_t init_conf(acl_conv_gemm_conf_t &acp, memory_desc_t &src_md,
     const auto acl_layout = is_nspc ? arm_compute::DataLayout::NHWC
                                     : arm_compute::DataLayout::NCHW;
 
+    // int8?
+    acp.is_int8 = utils::one_of(src_d.data_type(), s8, u8)
+            && wei_d.data_type() == s8;
+
+    auto acl_src_data_t
+            = acl_gemm_convolution_utils::get_acl_data_t(src_d.data_type());
+    auto acl_wei_data_t
+            = acl_gemm_convolution_utils::get_acl_data_t(wei_d.data_type());
+    auto acl_dst_data_t
+            = acl_gemm_convolution_utils::get_acl_data_t(dst_d.data_type());
+
+    auto acl_bia_data_t = arm_compute::DataType::F32;
+    if (acp.is_int8) acl_bia_data_t = arm_compute::DataType::S32;
+
     // clang-format off
     acp.src_info = arm_compute::TensorInfo(
             is_nspc ? arm_compute::TensorShape(ic, iw, ih, mb) :
             arm_compute::TensorShape(iw, ih, ic, mb),
             1,
-            arm_compute::DataType::F32,
+            acl_src_data_t,
             acl_layout);
 
     acp.wei_info = arm_compute::TensorInfo(
             is_nspc ? arm_compute::TensorShape(ic, kw, kh, oc) :
             arm_compute::TensorShape(kw, kh, ic, oc),
             1,
-            arm_compute::DataType::F32,
+            acl_wei_data_t,
             acl_layout);
 
     acp.dst_info = arm_compute::TensorInfo(
             is_nspc ? arm_compute::TensorShape(oc, ow, oh, mb) :
             arm_compute::TensorShape(ow, oh, oc, mb),
             1,
-            arm_compute::DataType::F32,
+            acl_dst_data_t,
             acl_layout);
 
     acp.bia_info = arm_compute::TensorInfo(
             acp.with_bias ? arm_compute::TensorShape(oc)
                           : arm_compute::TensorShape(),
             1,
-            arm_compute::DataType::F32,
+            acl_bia_data_t,
             acl_layout);
     // clang-format on
 
+    // Add quantization info to tensors
+    const float *scales = attr.output_scales_.scales_;
```

Comment:
Very interesting! Thank you for explaining, I was under the impression that zero-points are supported at run-time only and didn't realise the two options were applicable for both os and zp. It's a lot clearer now so thanks again.

In regards to the Compute Library, it doesn't currently support runtime quantization info. So if I understand correctly, for my current code to work I need to filter out the second option by specifing the correct mask at init checks?
If that's all, it's fixed now.

Reply:
Right, there are some [limitations ](https://github.com/oneapi-src/oneDNN/blob/d33607d750f5ae2a34f38aab62cb64053df22d7e/src/common/primitive_attr.cpp#L91) on zero points:

- If it is a buffer of values then it is supported only at runtime;
- If it is a single value then both runtime and creation time options are supported.

It was implemented this way because:
- Keeping zero points buffers for multiple primitive arguments would make attributes pretty heavy;
- There are not many users of creation time scales / zero points

> In regards to the Compute Library, it doesn't currently support runtime quantization info. So if I understand correctly, for my current code to work I need to filter out the second option by specifing the correct mask at init checks?
If that's all, it's fixed now.

Sounds good